navigate-route/src/main/java/com/esri/arcgismaps/sample/navigateroute/MainActivity.kt

```diff
@@ -211,128 +208,123 @@ class MainActivity : AppCompatActivity() {
         createRouteGraphics(routeGeometry)
 
         // set up a simulated location data source which simulates movement along the route
-        val simulationParameters = SimulationParameters(
-            Clock.System.now(),
-            35.0,
-            5.0,
-            5.0
+        val simulationParameters = SimulationParameters(Clock.System.now(),
+            velocity = 35.0,
+            horizontalAccuracy = 5.0,
+            verticalAccuracy = 5.0
         )
 
-        val simulatedLocationDataSource = SimulatedLocationDataSource().apply {
-            setLocationsWithPolyline(routeGeometry, simulationParameters)
-        }
+        // create the simulated data source using the geometry and parameters
+        val simulatedLocationDataSource = SimulatedLocationDataSource(
+            routeGeometry, simulationParameters
+        )
 
         // set up a RouteTracker for navigation along the calculated route
         val routeTracker = RouteTracker(routeResult, 0, true)
 
+        // plays the direction voice guidance
+        lifecycleScope.launch {
+            updateVoiceGuidance(routeTracker)
+        }
```

Comment:
I am still seeing some problems with this if the user resets the route. I found that when starting the route, waiting a few seconds, and resetting it, I started receiving each piece of guidance twice. I think the job from this coroutine would need to be cancelled every time `startNavigation` is called, and same for the coroutine that updates the location text. Or alternatively is it possible to reset the route without recalculating it and creating a new route tracker?

Reply:
Noticed, I think canceling the coroutine would make the best sense here. Added the following code to the click listener: 
```kotlin
resetNavigationButton.setOnClickListener {
    lifecycleScope.launch {
        if (locationDisplayJob.isActive) {
            // stop location data sources
            simulatedLocationDataSource.stop()
            routeTrackerLocationDataSource.stop()
            // cancel the coroutine jobs
            locationDisplayJob.cancelAndJoin()
            autoPanModeJob.cancelAndJoin()
            // start navigation again
            startNavigation(routeResult)
        }
    }
}
```